### PR TITLE
added new command class ManufacturerProprietary for Fibaro FGRM-222

### PR DESCRIPTION
**Pull request by ChristianKuehnel**
As discussed some time ago in the [forum](https://groups.google.com/forum/#!topic/openzwave/NzgcUyCwp3I): The Fibaro FGRM-222 Roller Shutter 2 is using the `COMMAND_CLASS_MANUFACTURER_PROPRIETARY` to indicate the tilt angle of the slats. This is only transmitted when activating the venetian blind mode and setting the  report type to "Fibaro Command class".

The next step would then be to add the functions to set the tile angle and also to request it.

### CODE REVIEW

File: cpp/src/command_classes/ManufacturerProprietary.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+using namespace OpenZWave;
+
+enum ProprietaryCmd
+{
+	ProprietaryCmd_Set		= 0x01,
+	ProprietaryCmd_Get		= 0x02,
+	ProprietaryCmd_Report	= 0x03
+};
+
+const uint8 FIBARO[2] = { 0x01, 0x0f };
```
Comment (Fishwaldo) [inline cpp/src/command_classes/ManufacturerProprietary.cpp]: Please use a more descriptive name to identify this represents a Fibaro (or other manufacturer in the future)

File: cpp/src/command_classes/ManufacturerProprietary.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+const uint8 FIBARO_BLINDS_SLATS[3] = {0x26, 0x03, 0x03};
```
Comment (Fishwaldo) [inline cpp/src/command_classes/ManufacturerProprietary.cpp]: Again, a different name to indicate what it is. I have no idea what a BLINDS_STATS does/is

File: cpp/src/command_classes/ManufacturerProprietary.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	uint8 const* payload = _data+2;
+	if( FIBARO[0] == _data[0] &&
+			FIBARO[1] == _data[1] ){
+
+		if( FIBARO_BLINDS_SLATS[0] == payload[0] &&
+				FIBARO_BLINDS_SLATS[1] == payload[1] &&
+				FIBARO_BLINDS_SLATS[2] == payload[2] ){
+			uint8 blinds = payload[3];
+			uint8 slats = payload[4];
+			ValueByte* blindsValue = static_cast<ValueByte*>( GetValue( _instance, 0 ) );
```
Comment (Fishwaldo) [inline cpp/src/command_classes/ManufacturerProprietary.cpp]: Please use a enum for the Value Index's. I would like to not "reuse" index's later if different manufacturers implement this CC. So no one else should ever implement Index 1 or 2.

File: cpp/src/command_classes/ManufacturerProprietary.cpp (hunk 1 above)
Comment (ChristianKuehnel) [inline cpp/src/command_classes/ManufacturerProprietary.cpp]: fixed

File: cpp/src/command_classes/ManufacturerProprietary.cpp (hunk 2 above)
Comment (ChristianKuehnel) [inline cpp/src/command_classes/ManufacturerProprietary.cpp]: fixed

File: cpp/src/command_classes/ManufacturerProprietary.cpp (hunk 3 above)
Comment (ChristianKuehnel) [inline cpp/src/command_classes/ManufacturerProprietary.cpp]: fixed

**Comment by ChristianKuehnel** [comment]
@Fishwaldo I just did a rebase, but from looking at the number of changes here, I'm not sure if that was the right way to do a rebase...

**Comment by Fishwaldo** [comment]
You should put your changes on top of the dev branch.  Just changing the target in Github won't work.

**Comment by ChristianKuehnel** [comment]
I complelety messed up my branches. Closing this pull requests and creating a new, clean one...

**Pull request closed without merging**